tests/sensors/test_jira_issue_sensor.py

```diff
@@ -0,0 +1,105 @@
+    """
+    Test the poll() method of JiraIssueSensor to ensure that
+    it searches for the correct issues and dispatches triggers
+    for each item found.
+    """
+
+    # Setup mock JIRA issues that would be returned by search_issues
+    mock_issue_1 = MagicMock(key="ISSUE-101")
+    mock_issue_2 = MagicMock(key="ISSUE-202")
+    # The sensor calls search_issues for multiple request types:
+    # ["Request New Project", "Add User", ...].
+    # We'll just pretend both calls return the same set of issues.
+    mock_search_issues.return_value = [mock_issue_1, mock_issue_2]
+
+    # Run sensor setup, which initializes self.jira_client
+    sensor.setup()
+
+    # Call poll(), which should iterate over request types,
+    # call search_issues, and dispatch for each returned issue.
+    sensor.poll()
+
+    # --- Verify search_issues is called with the expected parameters. ---
+
+    # The sensor uses:
+    #    request_type_list = ["Request New Project", "Add User"]
+    # so search_issues should be called once per request type.
+    # For "Request New Project":
+    expected_req_list_new_project = [
+        'statusCategory in ("Ready For Automation")',
+        '"Request Type" = "Request New Project"',
+    ]
+
+    # For "Add User":
+    expected_req_list_add_user = [
+        'statusCategory in ("Ready For Automation")',
+        '"Request Type" = "Add User"',
+    ]
+
+    # You can assert that these calls happen in any order
+    mock_search_issues.assert_any_call(
+        JiraAccount, "STFCCLOUD", expected_req_list_new_project
+    )
+    mock_search_issues.assert_any_call(
+        JiraAccount, "STFCCLOUD", expected_req_list_add_user
+    )
+
+    # --- Verify dispatch calls ---
+    # We expect two calls per request type because we have two mock issues.
+    # For "Request New Project" => trigger = "jira.request_new_project"
+    # For "Add User" => trigger = "jira.add_user"
+    # You can assert these calls in any order.
+
+    # Check that each issue resulted in a dispatch call for "Request New Project"
+    sensor.sensor_service.dispatch.assert_any_call(
+        trigger="jira.request_new_project", payload={"issue_key": "ISSUE-101"}
+    )
+    sensor.sensor_service.dispatch.assert_any_call(
+        trigger="jira.request_new_project", payload={"issue_key": "ISSUE-202"}
+    )
+
+    # Check that each issue resulted in a dispatch call for "Add User"
+    sensor.sensor_service.dispatch.assert_any_call(
+        trigger="jira.add_user", payload={"issue_key": "ISSUE-101"}
+    )
+    sensor.sensor_service.dispatch.assert_any_call(
+        trigger="jira.add_user", payload={"issue_key": "ISSUE-202"}
+    )
+
+    # Optionally, ensure total dispatch calls match the expected count:
+    # 2 issues * 2 request types = 4 dispatches
+    assert sensor.sensor_service.dispatch.call_count == 4
+
+    # If you want to confirm the JIRA client was correctly created in setup():
+    mock_jira_client.assert_called_once()
```

Comment:
Ah actually it's hidden at the bottom, even more reason for a dedicated test case :wink: